python/metrics/src/hydrotools/metrics/metrics.py

```diff
@@ -112,6 +113,62 @@ def nash_sutcliffe_efficiency(
         return 1.0 / (1.0 + numerator/denominator)
     return 1.0 - numerator/denominator
 
+def kling_gupta_efficiency(
+    y_true: npt.ArrayLike,
+    y_pred: npt.ArrayLike,
+    r_scale: float = 1.0,
+    a_scale: float = 1.0,
+    b_scale: float = 1.0
+    ) -> float:
+    """Compute the Kling-Gupta model efficiency coefficient (KGE).
+        
+    Parameters
+    ----------
+    y_true: array-like of shape (n_samples,) or (n_samples, n_outputs)
+        Ground truth (correct) target values, also called observations, measurements, or observed values.
+    y_pred: pandas.Series, required
+        Estimated target values, also called simulations or modeled values.
+    r_scale: float, optional, default 1.0
+        Linear correlation (r) scaling factor. Used to re-scale the Euclidean space by 
+        emphasizing different KGE components.
+    a_scale: float, optional, default 1.0
+        Relative variability (alpha) scaling factor. Used to re-scale the Euclidean space by 
+        emphasizing different KGE components.
+    b_scale: float, optional, default 1.0
+        Relative mean (beta) scaling factor. Used to re-scale the Euclidean space by 
+        emphasizing different KGE components.
+        
+    Returns
+    -------
+    score: float
+        Kling-Gupta efficiency.
+        
+    References
+    ----------
+    Gupta, H. V., Kling, H., Yilmaz, K. K., & Martinez, G. F. (2009). Decomposition of 
```

Comment:
on a side note. it would be "cool" if you could get the citation for the paper in bibtex or some other format. I am thinking a decorator that wraps each metric function and contains the bibtex formatted citation. You could access it like:

```python
from hydrotools.metrics import metrics

print(metrics.kling_gupta_efficiency.citation(format="bibtex"))

@article{Gupta2009,
  doi = {10.1016/j.jhydrol.2009.08.003},
  url = {https://doi.org/10.1016/j.jhydrol.2009.08.003},
  year = {2009},
  month = oct,
  publisher = {Elsevier {BV}},
  volume = {377},
  number = {1-2},
  pages = {80--91},
  author = {Hoshin V. Gupta and Harald Kling and Koray K. Yilmaz and Guillermo F. Martinez},
  title = {Decomposition of the mean squared error and {NSE} performance criteria: Implications for improving hydrological modelling},
  journal = {Journal of Hydrology}
}
```

Reply:
added

Reply:
That would be cool. Let's discuss that at next hydrotools meeting!